formats/src/main/scala/overflowdb/formats/dot/DotExporter.scala

```diff
@@ -0,0 +1,105 @@
+package overflowdb.formats.dot
+
+import overflowdb.formats.{ExportResult, Exporter, iterableForList}
+import overflowdb.{Edge, Graph, Node}
+
+import java.nio.file.{Files, Path}
+import scala.jdk.CollectionConverters.{IterableHasAsScala, MapHasAsScala}
+import scala.util.Using
+
+/**
+ * Exports OverflowDB Graph to graphviz dot/gv file
+ *
+ * Note: GraphML doesn't natively support list property types, so we fake it by encoding it as a `;` delimited string.
+ * If you import this into a different database, you'll need to parse that separately.
+ *
+ * https://en.wikipedia.org/wiki/DOT_(graph_description_language)
+ * https://www.graphviz.org/doc/info/lang.html
+ * http://magjac.com/graphviz-visual-editor/
+ * https://www.slideshare.net/albazo/graphiz-using-the-dot-language
+ * */
+object DotExporter extends Exporter {
+
+  override def runExport(graph: Graph, outputRootDirectory: Path) = {
+    val outFile = resolveOutputFile(outputRootDirectory)
+    var nodeCount, edgeCount = 0
+
+    Using.resource(Files.newBufferedWriter(outFile)) { writer =>
+      def writeLine(line: String): Unit = {
+        writer.write(line)
+        writer.newLine()
+      }
+
+      writeLine("digraph {")
+
+      graph.nodes().forEachRemaining { node =>
+        nodeCount += 1
+        writeLine(node2Dot(node))
+      }
+
+      graph.edges().forEachRemaining { edge =>
+        edgeCount += 1
+        writeLine(edge2Dot(edge))
+      }
+
+      writeLine("}")
+      writer.flush()
+      writer.close()
```

Comment:
Technically you don't have to `close` since it's withing `Using.resource` block

Reply:
very true, deleted, including the `flush` which happens on close anyway